### PR TITLE
Build API fetch function and implement in routes + testing

### LINKED ISSUE #6: Build API.ts
#### Create API functions

Create all necessary functions to get the requested data from API.

#### To Reproduce

[Sample](https://github.com/lmachens/photoplay/pull/49/commits/cb547e8bace337d68a1385fdc2c438ae314e2974)

#### Tasks

- [x] Create BASE_URL
- [x] Create fetch function
- [x] Create specific functions for API search

### PR DESCRIPTION
**Pull request by Niklas-Schaefers**
fixes #6

Not sure about:
https://github.com/Niklas-Schaefers/gitEncy/blob/e881dbe07e09af04dcb78e3bc578d76c24ff35be/src/server/routes.ts#L17

### CODE REVIEW

**Review by Dev-dfm — APPROVED** [review]
lgtm 👍

**Review by roabram — APPROVED** [review]
lgtm (:

File: src/server/gitHubSearch.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+type ErrorResult = {
+  message: string;
+};
+
+export async function fetchGitHubSearchCodeWithUser<T>(
+  code: string,
+  user: string
+): Promise<T> {
+  let url = "https://api.github.com/search/code?q=";
+  if (code && user) {
```
Comment (lmachens) [inline src/server/gitHubSearch.ts]: `code` and `user` are always set. I don't think you need this statement.

File: src/server/gitHubSearch.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import fetch from "node-fetch";
+
+type ErrorResult = {
+  message: string;
+};
+
+export async function fetchGitHubSearchCodeWithUser<T>(
```
Comment (lmachens) [inline src/server/gitHubSearch.ts]: This generic `<T>` is unused and doesn't make sense here.
Instead, try to create the return type based on the structure you get from GitHub (or don't add a return type for now).

File: src/server/routes.ts [hunk 3]
```diff
@@ -1,9 +1,27 @@
 import express from "express";
+import { fetchGitHubSearchCodeWithUser } from "./gitHubSearch";
 
 const router = express.Router();
 
-router.get("/search", (_req, res) => {
-  res.status(404).send("Hello World");
+router.get("/search", async (_req, res, next) => {
```
Comment (lmachens) [inline src/server/routes.ts]:
```suggestion
router.get("/search", async (req, res, next) => {
```

The underscore is only required for unused parameters.

File: src/server/routes.ts [hunk 4]
```diff
@@ -1,9 +1,27 @@
...
+    const { query } = _req;
+    if (!query.code) {
+      res.status(400).send("Code required");
+      return;
+    }
+    if (!query.user) {
+      res.status(400).send("User required");
+      return;
+    }
+    if (typeof query.code !== "string" || typeof query.user !== "string") {
```
Comment (lmachens) [inline src/server/routes.ts:17]: Good job. `query.code` could be an array if you add it multiple times to your request like:
`/api/search?code=abc&code=123`

**Review by lmachens — COMMENTED** [review]
Good job

**Pull request merged**